### PR TITLE
docs: add beehive document

### PR DESCRIPTION
**Pull request by sids-b**
add beehive document for understanding functions of beehive

### CODE REVIEW

**Comment by sids-b** [comment]
@rohitsardesai83 @m1093782566 , please review.

File: docs/modules/beehive.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+4) **typeChsLock:** lock for typeChannels map 
+5) **anonChannels:** anonChannels is a map of string(parentid) to chan(value) of message which will be used for sending response for a sync message.
+6) **anonChsLock:** lock for anonChannels map
+
+## Module Operations 
+### Add Module
+1) Add module operation first creates a new channel of message type.
+2) Then the module name(key) and its channel(value) is added in the channels map of channel context structure. 
+3) Eg: add edged module
+```shell
```
Comment (rohitsardesai83) [inline docs/modules/beehive.md]: nit: shell --> go

File: docs/modules/beehive.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  1) Header :  
+       1) ID : message ID (string)
+       2) ParentID : if it is a response to a sync message then parentID exists (string)
+       3) TimeStamp : time when message was generated (int)
+       4) Sync : flag to indicate if message is of type sync (bool)
+  2) Route : 
+       1) Source : origin of message (string)
+       2) Group : where message has to be broadcasted (string)
+       3) Operation : what’s the operation on the resource (string)
+       4) Resource : what’s the resource want to operate on (string)
```
Comment (rohitsardesai83) [inline docs/modules/beehive.md]: nit: the description could be changed to :     **the resource to operate on.**

File: docs/modules/beehive.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+## Message Format
+Message has 3 parts 
+  1) Header :  
+       1) ID : message ID (string)
+       2) ParentID : if it is a response to a sync message then parentID exists (string)
+       3) TimeStamp : time when message was generated (int)
+       4) Sync : flag to indicate if message is of type sync (bool)
+  2) Route : 
+       1) Source : origin of message (string)
+       2) Group : where message has to be broadcasted (string)
```
Comment (rohitsardesai83) [inline docs/modules/beehive.md]: the group to which the message has to be broadcasted.

File: docs/modules/beehive.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+Beehive support following message operations : 
+1) Send to a module/group
+2) Receive by a module
+3) Send Sync to a module/group
+4) Send Response to a sync message
+
+## Message Format
+Message has 3 parts 
+  1) Header :
```
Comment (m1093782566) [inline docs/modules/beehive.md]: nit: `FOO : BAR` -> `FOO: BAR`? I mean remove the white space prior to `:`?

There are many similar places below.

File: docs/modules/beehive.md (hunk 2 above)
Comment (sids-b) [reply docs/modules/beehive.md]: Done

File: docs/modules/beehive.md (hunk 3 above)
Comment (sids-b) [reply docs/modules/beehive.md]: Done

File: docs/modules/beehive.md (hunk 4 above)
Comment (sids-b) [reply docs/modules/beehive.md]: Done.

File: docs/modules/beehive.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+```go.
+err := coreContext.Send2GroupSync("edged",message,60*time.Second)
+```
+
+### Send Resp to a sync message
+1) SendResp is used to send response for a sync message.
+2) The messageID for which response is sent needs to be in the parentID of the response message.
+3) When SendResp is called, it checks if for the parentID of response message , there exists a channel is anonChannels.
+4) If channel exists, message(response) is sent on that channel.
+5) Orelse error is logged.
```
Comment (m1093782566) [inline docs/modules/beehive.md]: nit: Or else?

**Pull request merged**